tests/system/Pager/PagerTest.php

```diff
@@ -182,6 +182,22 @@ public function testStoreWithSegments()
         );
     }
 
+    public function testStoreWithURIReturnObject()
```

Comment:
What is this test testing?

Reply:
For covered testing if `$returnObject = true`

```php
$this->pager->store('bar', 5, 25, 100, 1);
$uri = $this->pager->getPageURI(7, 'bar', true);
```

Reply:
Why don't you assert that `$uri` is instance of `URI`?

Reply:
I call `getTotalSegments` and `getSegment` because part of the function URI class. Its okay if only assert instance of `URI`

Reply:
Please try to write a test that is as clear as possible what you are trying to test.

And if you want to test two aspects, you can add two test methods.

Reply:
What about this?

```diff
--- a/tests/system/Pager/PagerTest.php
+++ b/tests/system/Pager/PagerTest.php
@@ -184,11 +184,28 @@ final class PagerTest extends CIUnitTestCase

     public function testGetPageURIWithURIReturnObject()
     {
-        $this->pager->store('bar', 5, 25, 100, 1);
+        $_GET['page'] = 3;
+        $_GET['foo']  = 'bar';
+
+        $this->pager->store('default', 3, 25, 100, 1);

-        $uri = $this->pager->getPageURI(7, 'bar', true);
+        $uri = $this->pager->getPageURI(5, 'default', true);

         $this->assertInstanceOf(URI::class, $uri);
+        $this->assertSame(
+            'page=3&foo=bar',
+            $uri->getQuery()
+        );
+
+        $this->pager->store('new', 3, 25, 100, 1);
+
+        $uriOnly = $this->pager->only(['foo'])->getPageURI(7, 'new', true);
+
+        $this->assertInstanceOf(URI::class, $uriOnly);
+        $this->assertSame(
+            'foo=bar',
+            $uriOnly->getQuery()
+        );
     }
```

Reply:
It is better than the first test.
But there are many test aspects:
- return URI object
- query parameters in URI object
- only() method

It is better to write one test for one aspect.

If the test method is `testGetPageURIWithURIReturnObject`, the current test is enough.
If you want to write a test for only() or query parameters, please add another test method.
Or is there any test already?

Reply:
`getPageURI` only 2 aspect, `with only` or `without only`. And this topic coverage `getPageURI` return `URI` object

```php
    public function testGetPageURIWithURIReturnObject()
    {
        $_GET['page'] = 3;
        $_GET['foo']  = 'bar';

        $this->pager->store('default', 5, 25, 100, 1);

        $uri = $this->pager->getPageURI(5, 'default', true);

        $this->assertInstanceOf(URI::class, $uri);
        $this->assertSame(
            'page=3&foo=bar',
            $uri->getQuery()
        );
    }

    public function testOnlyGetPageURIWithURIReturnObject()
    {
        $_GET['page'] = 3;
        $_GET['foo']  = 'bar';

        $this->pager->store('default', 5, 25, 100, 1);

        $uriOnly = $this->pager->only(['foo'])->getPageURI(5, 'default', true);

        $this->assertInstanceOf(URI::class, $uriOnly);
        $this->assertSame(
            'foo=bar',
            $uriOnly->getQuery()
        );
    }
```